Refactor helper scripts for infrastructure, jumphost and artifactory.

terror96 opened the pull request:
Refactor the following scripts for managing:

* infrastructure
* jumphost
* artifactory

Restructure the infra configuration file.

Added common helpers for parsing command line arguments.

Updated documentation for infrastructure, jumphost and artifactory management.

tuminoid commented on ci/scripts/common/utils.sh:
```diff
@@ -0,0 +1,18 @@
+COMMON_SCRIPTS_DIR="${CI_DIR}/scripts/common"
+
+# shellcheck source=ci/scripts/common/opts.sh
+source "${COMMON_SCRIPTS_DIR}/opts.sh"
```
.

tuminoid commented on ci/scripts/jumphost/README.md:
```diff
@@ -1,71 +1,210 @@
+$ ./create_or_rebuild_jumphost.sh
 
-The public key should be given as content, not as a file.
+Usage:
```
Drop

tuminoid commented on ci/scripts/jumphost/README.md:
```diff
@@ -1,71 +1,210 @@
-The scripts in this folder are aimed at managing the dev environment jumphost
+Author: [Tero Kauppinen](mailto:tero.kauppinen@est.tech)
+\
+Last update: 10.9.2025
```
Drop

tuminoid commented on ci/scripts/jumphost/README.md:
````diff
@@ -1,71 +1,210 @@
+```
 
-Some environment variables need to be set
+It should be noted that in this context, *key-file* and *user*
````
Backticks

tuminoid commented on ci/scripts/jumphost/README.md:
```diff
@@ -1,71 +1,210 @@
+$ ./update_jumphost_users.sh
 
-   ./delete_dev_jumphost.sh
+Usage:
```
Drop

tuminoid commented on ci/scripts/jumphost/README.md:
````diff
@@ -1,71 +1,210 @@
+use the following script:
+
+```bash
+$ ./add_jumphost_user.sh
````
Drop

tuminoid commented on ci/scripts/jumphost/README.md:
````diff
@@ -1,71 +1,210 @@
+```bash
+$ ./del_jumphost_user.sh
+
+Usage:
````
Drop

tuminoid commented on ci/scripts/jumphost/add_jumphost_user.sh:
```diff
@@ -1,30 +1,27 @@
 #
 # Usage:
-#   add_jumphost_user.sh <user_name> <file_path_containing_all_user_keys>
+#   add_jumphost_user.sh
```
Why remove the args?

terror96 replied:
Mistake.

tuminoid commented on ci/scripts/jumphost/add_jumphost_user.sh:
```diff
@@ -1,30 +1,27 @@
-USER_AUTHORIZED_KEYS_FILE="${2:?}"
-
+# shellcheck source=ci/scripts/common/opts.sh
+source "${COMMON_SCRIPTS_DIR}/opts.sh"
+# shellcheck source=ci/scripts/common/utils.sh
+source "${COMMON_SCRIPTS_DIR}/utils.sh"
 # shellcheck source=ci/scripts/artifactory/utils.sh
 source "${RT_SCRIPTS_DIR}/utils.sh"
```
.

tuminoid commented on ci/scripts/jumphost/create_or_rebuild_jumphost.sh:
Same nits

tuminoid commented on ci/scripts/jumphost/del_jumphost_user.sh:
Same nits

tuminoid commented on ci/scripts/jumphost/delete_jumphost.sh:
Same

tuminoid commented on ci/scripts/jumphost/files/add_proxy_user.sh:
```diff
@@ -43,16 +41,22 @@ Match GROUP ${PROXY_USERS_GROUP}
 
-  sudo systemctl restart sshd
+  # TODO: revise the logic to restart the SSH daemon
+  if [[ -r "/usr/lib/systemd/system/ssh.service" ]]
```
We should not be checking existence of a file, but something like `systemctl is-active`, but TODO is in place.

tuminoid reviewed (commented):
Very good improvements, just some repeating nits in the readme files and in the scripts. Also some future items noted to clean these up. Some constructs used for no prupose IMO, let's stick to simpler more maintanable code.

Stopped repeating `same` halfway through :) 
I'll take another look after next revision.

terror96 commented on ci/scripts/common/opts.sh:
```diff
@@ -0,0 +1,330 @@
+
+  # Print out the content of the option array and print only
+  # selected options
+  for ((I=1; I<$#+1; I++)); do
```
If you mean:

common_make_usage_string --one-liner "ONE-LINER" --arguments "ARGUMENTS" <option contants>

Sure, it is just more code.

terror96 commented on ci/scripts/common/opts.sh:
```diff
@@ -0,0 +1,330 @@
+# common_validate_target
+#
+common_validate_target() {
+  [[ -z "${COMMON_OPT_TARGET_VALUE}" ]] && {
```
Has been a common practice, but can use a longer as well. However, the parts of the scripts that I did not refactor also use the sort form.

terror96 commented on ci/scripts/artifactory/utils.sh:
```diff
@@ -20,6 +30,7 @@ rt_upload_artifact() {
 
+  common_verbose "Upload artifact: ${SRC_PATH},${DST_PATH},${ANONYMOUS}"
   _CMD="curl \
     $( ([[ "${ANONYMOUS}" != 1 ]] && echo " -u${RT_USER:?}:${RT_TOKEN:?}") || true) \
```
Agreed.

tuminoid reviewed (commented):
We reviewed this offline with demo included, @terror96 has notes what has to be fixed.
/hold

Bash cleanup stuff is agreed to be done in follow-up in order to keep this sane size.

terror96 commented:
With the new version, I added "dry-run" and merged all user information scripts into a single script, and hopefully addressed the issues that popped up during testing with @tuminoid.

terror96 commented on ci/scripts/jumphost/sync_jumphost_users.sh:
```diff
@@ -0,0 +1,187 @@
+  common_verbose "Copy SSH key ${USER_AUTHORIZED_KEYS_FILE} to"\
+                 "${JUMPHOST_PUBLIC_IP}:/tmp/${USER}_auth_keys..."
+
+  common_run -- scp \
```
Makes sense. Let me do the adjustment already in this PR...

tuminoid reviewed (commented):
This is coming along nicely, with a list of improvements, cleanups and simplifications to be done in subsequent PRs. No major complaints left, so
/approve
